o2a/mappers/shell_mapper.py

```diff
@@ -35,48 +35,61 @@ def __init__(
         self,
         oozie_node: Element,
         name: str,
+        job_properties: Dict[str, str],
+        configuration_properties: Dict[str, str],
         trigger_rule: str = TriggerRule.ALL_SUCCESS,
-        params: Dict[str, str] = None,
         **kwargs,
     ):
-        ActionMapper.__init__(self, oozie_node=oozie_node, name=name, trigger_rule=trigger_rule, **kwargs)
-        if params is None:
-            params = {}
-        self.params = params
+        ActionMapper.__init__(
+            self,
+            oozie_node=oozie_node,
+            name=name,
+            trigger_rule=trigger_rule,
+            job_properties=job_properties,
+            configuration_properties=configuration_properties,
+            **kwargs,
+        )
         self._parse_oozie_node()
 
     def _parse_oozie_node(self):
         res_man_text = self.oozie_node.find("resource-manager").text
         name_node_text = self.oozie_node.find("name-node").text
-        self.resource_manager = el_utils.replace_el_with_var(res_man_text, params=self.params, quote=False)
-        self.name_node = el_utils.replace_el_with_var(name_node_text, params=self.params, quote=False)
-        self._parse_config()
+        self.resource_manager = el_utils.replace_el_with_var(
+            res_man_text, property_set=self.property_set, quote=False
+        )
+        self.name_node = el_utils.replace_el_with_var(
+            name_node_text, property_set=self.property_set, quote=False
+        )
         cmd_node = self.oozie_node.find("exec")
         arg_nodes = self.oozie_node.findall("argument")
         cmd = " ".join([cmd_node.text] + [x.text for x in arg_nodes])
         self.bash_command = el_utils.convert_el_to_jinja(cmd, quote=False)
         self.pig_command = f"sh {shlex.quote(self.bash_command)}"
 
-    def to_tasks_and_relations(self):
-        prepare_command = self.get_prepare_command(self.oozie_node, self.params)
-        tasks = [
-            Task(
-                task_id=self.name + "_prepare",
-                template_name="prepare.tpl",
-                template_params=dict(prepare_command=prepare_command),
+    def to_tasks_and_relations(self) -> Tuple[List[Task], List[Relation]]:
```

Comment:
The parent class have a type defintion. It is neccesary?

Reply:
Same as above. I prefer to leave it for reader's clarity..

Reply:
Please @mik-laj resolve this if you think the discussion is resolved.